### PR TITLE
add promoted-image-governor/README.md

### PR DESCRIPTION
**Pull request by hongkailiu**
/cc @openshift/test-platform

### CODE REVIEW

File: cmd/promoted-image-governor/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# promoted-image-governor
+
+## What it does
+
+`promoted-image-governor` is a tool with the following features:
+
+- For any image stream on every build-farm cluster, if it contains a tag promoted by some ci-operator's configuration, then
+  every tag in the stream has to be promoted too. Any violated tag will be deleted by this tool.
+- Generate [the image mapping files](https://github.com/openshift/release/tree/master/core-services/image-mirroring/openshift) for OpenShift/origin.
```
Comment (petr-muller) [inline cmd/promoted-image-governor/README.md:9]: How about "Generates the [mirroring mapping files] for quay.io/openshift organization"?

File: cmd/promoted-image-governor/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+- Explain why an `imagestreamtag` exists.
+
+
+## Why it exists
+
+- Regulate the image streams with promoted tags.
+- Reduce the manual work on the maintenance of the mapping files.
+- Save the effort of reversing engineering.
```
Comment (petr-muller) [inline cmd/promoted-image-governor/README.md:17]: "reverse engineering".

Also, reverse engineering of what?

File: cmd/promoted-image-governor/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+- For any image stream on every build-farm cluster, if it contains a tag promoted by some ci-operator's configuration, then
+  every tag in the stream has to be promoted too. Any violated tag will be deleted by this tool.
+- Generate [the image mapping files](https://github.com/openshift/release/tree/master/core-services/image-mirroring/openshift) for OpenShift/origin.
+- Explain why an `imagestreamtag` exists.
+
+
+## Why it exists
+
+- Regulate the image streams with promoted tags.
+- Reduce the manual work on the maintenance of the mapping files.
```
Comment (petr-muller) [inline cmd/promoted-image-governor/README.md:16]: ...and enforce their correctness

File: cmd/promoted-image-governor/README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# promoted-image-governor
+
+## What it does
+
+`promoted-image-governor` is a tool with the following features:
+
+- For any image stream on every build-farm cluster, if it contains a tag promoted by some ci-operator's configuration, then
```
[the remark this answers is not in the document]
Comment (petr-muller) [reply cmd/promoted-image-governor/README.md:7]: Also, not just build-farm clustrer but also (and I guess mainly) app.ci registry

File: cmd/promoted-image-governor/README.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  every tag in the stream has to be promoted too. Any violated tag will be deleted by this tool.
+- Generate [the image mapping files](https://github.com/openshift/release/tree/master/core-services/image-mirroring/openshift) for OpenShift/origin.
+- Explain why an `imagestreamtag` exists.
+
+
+## Why it exists
+
+- Regulate the image streams with promoted tags.
```
Comment (petr-muller) [inline cmd/promoted-image-governor/README.md:15]: I'm not sure about the "regulate" here. Regulate towards what goal? I think the main purpose is to delete the stale images that were promoted in the past but are no more, no?

**Review by petr-muller — COMMENTED** [review]
Thanks so much for writing the docs!

File: cmd/promoted-image-governor/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/promoted-image-governor/README.md:7]: > prevents a single tag from being promoted by two configs

I do not see it in this tool. But I did hear such a feature exists somewhere.

File: cmd/promoted-image-governor/README.md (hunk 1 above)
Comment (hongkailiu) [reply cmd/promoted-image-governor/README.md]: Updated.

File: cmd/promoted-image-governor/README.md (hunk 2 above)
Comment (hongkailiu) [reply cmd/promoted-image-governor/README.md]: Updated.

File: cmd/promoted-image-governor/README.md (hunk 3 above)
Comment (hongkailiu) [reply cmd/promoted-image-governor/README.md]: Updated.

File: cmd/promoted-image-governor/README.md (hunk 5 above)
Comment (hongkailiu) [reply cmd/promoted-image-governor/README.md]: Updated.

**Pull request merged**